cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2994,920 +3005,146 @@ endmacro()
 # Dependencies for Arrow Flight RPC
 
 macro(build_absl)
```

Comment:
Could you use `function()` not `macro()` to create a variable scope?

We can remove `cleanup_fetchcontent()` by this change.

Reply:
That was a good idea, moving to function help me clean a bunch of things that were not necessary! Thanks for the suggestion @kou